src/interface/blas3_interface.hpp

```diff
@@ -83,6 +84,22 @@ typename executor_t::policy_t::event_t _gemm_backend(
     index_t _K, element_t _alpha, container_0_t a_, index_t _lda,
     container_1_t b_, index_t _ldb, element_t _beta, container_2_t _C,
     index_t _ldc, index_t batch_size) {
+  if (_alpha == element_t{0}) {
+    // When alpha = 0, GEMM is equivalent to C = beta * C.
+    if (_ldc == _M) {
+      // When LDC is M, we can scale the full matrix at once.
+      return ::blas::_scal(ex, _N * _M, _beta, _C, index_t{1});
```

Comment:
I just realized one thing. When alpha is zero and beta is zero and C is NaN the result according to the netlib gemm should be zero, but in this case, I think the result will be NaN. maybe we need to add another check saying that if beta is zero as well, just sycl.memset(0);

Reply:
I added this change to `_scal`. If you would prefer this behaviour only for `_gemm` then I can move the change there.

Reply:
Thanks! I don't mind where it is added. Inside the scal is good.